frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/utils.ts

```diff
@@ -0,0 +1,54 @@
+import type { RequestorState } from "../../reducer";
+
+/**
+ * Get the value of the request body, if any. It doesn't support multi-part
+ * forms using `File` objects.
+ */
+export function getBodyValue({
+  body,
+  method,
+}: Pick<RequestorState, "body" | "method">) {
+  // Prevent sending JSON body for GET and HEAD requests, as they're the only
+  // methods where providing data is invalid.
+  if (method === "GET" || method === "HEAD" || !body.value) {
+    return undefined;
+  }
+
+  const bodyValue = (() => {
+    switch (body.type) {
+      case "form-data": {
+        return body.value.reduce((acc, field) => {
```

Comment:
so instead of constructing this manually, you could also make use of `URLSearchParams` and that would help you get encoding for free. i should've thought of this when we talked yesterday but it just hit me 

```ts
import { FormDataParameter } from "../../FormDataForm/types";

export function formDataToUrlEncoded(formDataParams: FormDataParameter[]) {
  const urlEncodedFormData = new URLSearchParams();

  formDataParams.forEach((field) => {
    if (field.enabled && field.value.type === "text") {
      urlEncodedFormData.append(field.key, field.value.value);
    }
  });

  return urlEncodedFormData.toString();
}
```

Reply:
then in your switch case just call `formDataToUrlEncoded(body.value)`

Reply:
This indeed is a nicer way to approach it. Will add!